src/reduced_basis/rb_eim_construction.C

```diff
@@ -54,6 +54,54 @@
 namespace libMesh
 {
 
+namespace
+{
+// We make an anonymous namespace for local helper functions. The helper functions below
+// are used to do some basic operators for QpDataMap and SideQpDataMap.
+
+// Recall that we have:
+// typedef std::map<dof_id_type, std::vector<std::vector<Number>>> QpDataMap;
+// typedef std::map<std::pair<dof_id_type,unsigned int>, std::vector<std::vector<Number>>> SideQpDataMap;
+
+// Implement u <- u + k*v
+template <typename DataMap>
+void add(DataMap & u, const Number k, const DataMap & v)
+{
+  for (auto & [key,val] : u)
```

Comment:
This is passing all the CI tests, so @roystgnr do we officially allow structured bindings now? I thought it was a C++17 thing, and I can't remember if we went to C++14 or jumped all the way to C++17.

Reply:
We're *almost* ready to officially allow *basic* C++17 structured bindings.  Personally I'd rather wait until we have proper CI going before letting them in.

The catches here are:

1. We're not ready yet, only *almost* ready.  We're literally getting new module trees built with new min-gcc and min-clang compilers right now.  Currently CI is testing with a min-clang that handles (and so enables because I foolishly didn't ask it to go lower) C++17, and we're not testing with a min-gcc (because the only older module we already had handy handles C++11 but not C++14), so we get false positives here.

2. We'll be getting *basic* C++17 with our min-clang, not the handful of Defect Reports that have led to changes in the standard since then, one of which is the way structured bindings work (or, in the original standard, fail to work) with non-public member variables.  Not an issue here, and I'd figured it wouldn't be an issue for a while, but if we're chomping at the bit to use structured bindings then maybe I made the wrong call?

Reply:
I pushed an update which removed the structured binding, so hopefully that resolves this.